migration-engine/connectors/sql-migration-connector/src/sql_migration_persistence.rs

```diff
@@ -183,26 +186,41 @@ async fn last_applied_migrations(
 }
 
 fn migration_table_setup_sqlite(t: &mut barrel::Table) {
-    migration_table_setup(t, types::date(), types::custom("TEXT"));
+    migration_table_setup(t, types::text(), types::date(), types::custom("TEXT"));
 }
 
 fn migration_table_setup_postgres(t: &mut barrel::Table) {
-    migration_table_setup(t, types::custom("timestamp(3)"), types::custom("TEXT"));
+    migration_table_setup(t, types::text(), types::custom("timestamp(3)"), types::custom("TEXT"));
 }
 
 fn migration_table_setup_mysql(t: &mut barrel::Table) {
-    migration_table_setup(t, types::custom("datetime(3)"), types::custom("LONGTEXT"));
+    migration_table_setup(
+        t,
+        types::text(),
+        types::custom("datetime(3)"),
+        types::custom("LONGTEXT"),
+    );
+}
+
+fn migration_table_setup_mssql(t: &mut barrel::Table) {
+    migration_table_setup(
+        t,
+        types::custom("nvarchar(max)"),
+        types::custom("datetime2"),
+        types::custom("nvarchar(max)"),
+    );
 }
 
 fn migration_table_setup(
```

Comment:
Reminder for @tomhoule to check this in the final review.

Reply:
lgtm